### PR TITLE
Address regression caused by chef client 12.5 environment argument

### PR DESCRIPTION
**Pull request by adamedx**
I believe this is the commit in Chef (release in 12.5.1) that caused the regression in knife-windows: https://github.com/chef/chef/commit/14d7baab6365e0b4146990a5306f80a7f0d09a82 -- cc @danielsdeleo. I'm just not exactly sure how...

### CODE REVIEW

**Comment by mwrock** [comment]
You may have already caught on to this, but you will need to update the support templates and remove the `-E _defaut`

**Comment by adamedx** [comment]
Yeah, in the process of adding some 12.5 specific specs.

**Comment by adamedx** [comment]
@mwrockx, @btm, @chefsalim can you take a look?

**Comment by mwrock** [comment]
looks good but I don't think you meant to add the ~ files. :+1:  after those get deleted.

**Comment by adamedx** [comment]
What `~` files? :smile: Thank you @mwrock!

**Comment by mwrock** [comment]
Are you using MS Word as an editor?

**Comment by adamedx** [comment]
hehe, just emacs, which will give you files with `~`, `#`, and sometimes both. Should probably update .gitignore...

**Comment by mwrock** [comment]
:+1:  again.

FYI @btm @chefsalim  bootstraps are broken until this is merged. I had to manually add the fix while testing the policyfile support to get it to work. Would bwe greato to get another +1 soon

**Comment by danielsdeleo** [comment]
:+1: :shipit:

File: lib/chef/knife/core/windows_bootstrap_context.rb [hunk 1]
```diff
@@ -140,8 +140,9 @@ def config_content
         end
 
         def start_chef
+          bootstrap_environment_option = bootstrap_environment.nil? ? '' : " -E #{bootstrap_environment}"
           start_chef = "SET \"PATH=%PATH%;C:\\ruby\\bin;C:\\opscode\\chef\\bin;C:\\opscode\\chef\\embedded\\bin\"\n"
-          start_chef << "chef-client -c c:/chef/client.rb -j c:/chef/first-boot.json -E #{bootstrap_environment}\n"
+          start_chef << "chef-client -c c:/chef/client.rb -j c:/chef/first-boot.json#{bootstrap_environment_option}\n"
```
Comment (btm) [inline lib/chef/knife/core/windows_bootstrap_context.rb]: Does mean knife-windows requires Chef 12.5+ now? If so, can we say something in the README?

Comment (mwrock) [inline lib/chef/knife/core/windows_bootstrap_context.rb]: I think this is backwards compatible. Before if `-E` was ommited, we would add `-E -default` now we just omit it alltogether which I believe is effectively the same.

File: spec/functional/bootstrap_download_spec.rb [hunk 2]
```diff
@@ -169,11 +177,19 @@ def run_download_scenario
       @template_output = sample_data('win_template_rendered_with_bootstrap_install_command.txt')
     end
 
-    it "bootstrap_install_command option is rendered in the windows-chef-client-msi.erb template" do
+    it "bootstrap_install_command option is rendered in the windows-chef-client-msi.erb template", :chef_lt_12_5_only => true do
```
Comment (btm) [inline spec/functional/bootstrap_download_spec.rb]: it'd be kind of nice if these were contexts too, like the 12.5.0+ below.

**Comment by btm** [comment]
:+1:

**Pull request merged**